Automattic/jetpack
# Build: Prepare tooling for Gutenberg extensions in Jetpack

Adds files from Jetpack Gutenberg extensions folder (about to be merged in https://github.com/Automattic/jetpack/pull/11633) to ignore lists relevant for the packaged version of the plugin.

#### Changes proposed in this Pull Request:

* SVN Ignores
* CDN asset ignores
* Ignores for release branches

#### Testing instructions:
Together with https://github.com/Automattic/jetpack/pull/11633 (`git merge --squash --no-commit origin/add/jetpack-blocks-src`) — does the build ignore assets in `/extensions/*` but still sees `*.php` files?

- `yarn build-production`
- `php bin/build-asset-cdn-json.php`

An alternative way of testing this would be to:

- Checkout this branch. 
- Add the blocks src to: `git merge --squash --no-commit origin/add/jetpack-blocks-src`
- Comment out this line so we can build a release branch from our current branch:
  https://github.com/Automattic/jetpack/blob/e450f018a3612268e293d299a4f144948ddf3bc5/tools/build-release-branch.sh#L102
  You can apply this patch:
  https://gist.github.com/sirreal/2245987078998c2b6b9f95ba45d93d61#file-tooling-patch
- Commit the changes (`git add -A && git commit -am 'DROPME: testing changes'`)
- Run `tools/build-release-branch.sh -n`
- Pick a unique branch name.
- Let the build happen for both branches (the release branch and the built branch)
- Once the build is over, check the built branch
- It should not include any of the block's source files.
- the `.gitignore` should include the new lines.


#### Proposed changelog entry for your changes:

* None
